### PR TITLE
task 4.1

### PR DESCRIPTION
**Pull request by erinus1**
04cb198

### CODE REVIEW

File: src/Module4_1/Shape.java [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 public abstract class Shape {
 
-    public abstract void rectangleArea();
+    public abstract double rectangleArea(double length, double breadth);
 
-    public abstract void squeareArea();
+    public abstract double squeareArea(double side);
 
-    public abstract void circleArea();
+    public abstract double circleArea(double radius);
```
Comment (olelysieev-clgx) [inline src/Module4_1/Shape.java]: please try not to use empty lines

Comment (erinus1) [reply src/Module4_1/Shape.java]: Fixed

File: src/Module4_1/Area.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package Module4_1;
+
```
Comment (olelysieev-clgx) [inline src/Module4_1/Area.java:2]: Try to avoid such empty lines. Delete this one

File: src/Module4_1/Area.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    public class Area extends Shape {
+        @Override
+        public  double rectangleArea(double length, double breadth) {
+
+            return (length * breadth);
+        }
+
```
Comment (olelysieev-clgx) [inline src/Module4_1/Area.java:10]: empty line

File: src/Module4_1/Main.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+            if (breadth < 0) {
+                throw new Exception("Breadth must not be a negative one");
+            }
+
+            Area area = new Area();
+            System.out.println("Rectangle is " + area.rectangleArea(length, breadth));
+
+
+            System.out.println("Enter squeare's side: ");
```
Comment (olelysieev-clgx) [inline src/Module4_1/Main.java:31]: square's

File: src/Module4_1/Main.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            double side = in.nextDouble();
+
+
+            System.out.println("Squeare is " + area.squareArea(side));
```
Comment (olelysieev-clgx) [inline src/Module4_1/Main.java:35]: Square

File: src/Module4_1/Main.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                throw new Exception("Breadth must not be a negative one");
+            }
+
+            Area area = new Area();
+            System.out.println("Rectangle is " + area.rectangleArea(length, breadth));
+
+
+            System.out.println("Enter squeare's side: ");
+            double side = in.nextDouble();
+
```
Comment (olelysieev-clgx) [inline src/Module4_1/Main.java:33]: empty line

**Review by olelysieev-clgx — APPROVED** [review]

**Pull request merged**